### PR TITLE
vineyard: fix test on Linux

### PR DESCRIPTION
**Pull request by lukeshingles**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Test on Linux was broken by https://github.com/Homebrew/homebrew-core/pull/181351 (cannot find -lomp since llvm updated to 19.1.0)

### CODE REVIEW

File: Formula/v/vineyard.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class Vineyard < Formula
   url "https://github.com/v6d-io/v6d/releases/download/v0.23.2/v6d-0.23.2.tar.gz"
   sha256 "2a2788ed77b9459477b3e90767a910e77e2035a34f33c29c25b9876568683fd4"
   license "Apache-2.0"
-  revision 3
+  revision 4
```
Comment (carlocab) [inline Formula/v/vineyard.rb:9]:
```suggestion
  revision 3
```
Let's skip the `revision` bump here -- we don't need all the users of this formula to install an essentially identical version with `brew upgrade`.

**Comment by carlocab** [comment]
Also, note: since there is no Linux bottle (because the test failed when rebuilding it for `llvm@18`), CI will not show ❌ if there are failures on Linux. You'll need to open up the log to check if your changes worked.

**Comment by lukeshingles** [comment]
Ok, this seems ready to me. I removed the compiler overrides associated with this comment “macos AppleClang doesn't support -fopenmp”. Actually, Apple clang can use -fopenmp when libomp.rb is installed, but that wasn’t necessary here because vineyard only compiles with openmp on Linux.

**Review by carlocab — APPROVED** [review]
Thanks; great work!

**Pull request merged**